modules/api/pkg/provider/kubernetes/cluster.go

```diff
@@ -281,8 +281,17 @@ func (p *ClusterProvider) Get(ctx context.Context, userInfo *provider.UserInfo,
 		return nil, err
 	}
 	if options.CheckInitStatus {
-		if !cluster.Status.ExtendedHealth.AllHealthy() {
-			return nil, apierrors.NewServiceUnavailable("Cluster components are not ready yet")
+		healthCheck := cluster.Status.ExtendedHealth
+		if cluster.Spec.Cloud.ProviderName == string(kubermaticv1.EdgeCloudProvider) {
```

Comment:
I think it would be good if we add a comment here, why edgeprovider is different / why it's fine for edgeprovider if at least one of the components is healthy